docs/args-and-options.md

```diff
@@ -0,0 +1,180 @@
+
+The following methods construct an `Options` with a `name` that requires an input of the corresponding type from the user. Observe that for `Options` is necessary to specify the `name`.
+```scala mdoc:silent
+val name = "name"
+
+Options.boolean(name)
+Options.file(name)
+Options.directory(name)
+Options.text(name)
+Options.decimal(name)
+Options.integer(name)
+Options.instant(name)
+Options.localDate(name)
+Options.localDateTime(name)
+Options.localTime(name)
+Options.monthDay(name)
+Options.offsetDateTime(name)
+Options.offsetTime(name)
+Options.period(name)
+Options.year(name)
+Options.yearMonth(name)
+Options.zonedDateTime(name)
+Options.zoneId(name)
+Options.zoneOffset(name)
+```
+### Combining and transforming options
+We can use the following methods to create more complex Options:
+```scala mdoc:silent
+trait Options[A] {
+  def ++[A1 >: A, B](that: Options[B]): Options[(A1, B)]
+  def |[A1 >: A](that: Options[A1]): Options[A1]
+  def orElse[A1 >: A](that: Options[A1]): Options[A1] // same as |
+  def orElseEither[B](that: Options[B]): Options[Either[A, B]] 
+  def map[B](f: A => B): Options[B] 
+  def optional: Options[Option[A]] 
+
+  // Creates options with different aliases
+  def alias(name: String, names: String*): Options[A]
+
+  // Creates options that have a default value
+  def withDefault[A1 >: A](value: A1): Options[A1]
+}
+```
+
+### Example
+As an example, we are going to construct an instance of `Options[Order]` that defines two commands: ask for the path of a file and a string or a number to search in the file.
+```scala mdoc:silent:reset
+import zio.cli._
+import java.nio.file.{Path => JPath}
+
+// Construction of basic options.
+val getFile: Options[JPath] = Options.file("file")
+val getText: Options[String] = Options.text("textToSearch")
+val getInt: Options[BigInt]   = Options.integer("intToSearch")
+
+// First we construct a new Options[Either[String, Int]] that will allow to enter only one of the two possibilities in the CLI.
+val getTextOrInt: Options[Either[String, BigInt]] = getText orElseEither getInt
+
+// Then we are going to add some information that will appear if the user ask for help in the CLI App
+val doc: String = "String or Int to search in the specified file"
+val getTextOrIntWithDoc = getTextOrInt ?? doc
+
+// Finally we combine getFile with getTextOrInt 
+val options: Options[(JPath, Either[String, BigInt])] = getFile ++ getTextOrIntWithDoc
+
+```
+Note that we have not added any help to the `file` options or to `options` only to `getTextOrInt`.
+The type of Options is cumbersome to work with. We might want to transform that type. This can be done using method `map`. This method transform the output from a valid input using `f`.
+
+Defining a sealed trait to describe the output of the `Options` we can construct better `Options`.
+```scala mdoc:silent
+sealed trait Search
+case class StringSearch(file: JPath, text: String) extends Search
```

Comment:
For all of the example, I would try to make them as realistic as possible: so pick an existing command-line app that you are familiar with, and use examples for that.

Or, to make it more fun: pretend you and the reader are building a command-line app. You need to add options and arguments. So walk the user through the process of doing that, with good names and realistic types.